ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java

```diff
@@ -157,7 +158,7 @@ synchronized int process(Event event) {
   private final int maxOutstandingInstallSnapshots;
   private final TimeoutExecutor scheduler = TimeoutExecutor.getInstance();
 
-  private volatile StreamObservers appendLogRequestObserver;
+  private AtomicReference<StreamObservers> appendLogRequestObserver = new AtomicReference<>();
```

Comment:
Let's suppress the warning?  WRITE is protected by `writeLock` while READ is protected `volatile`.

Reply:
Brilliant design! Changed!